core/trino-main/src/main/java/io/trino/sql/gen/InCodeGenerator.java

```diff
@@ -91,10 +92,9 @@ public InCodeGenerator(SpecialForm specialForm)
     @VisibleForTesting
     static SwitchGenerationCase checkSwitchGenerationCase(Type type, List<RowExpression> values)
     {
-        if (values.size() > 32) {
-            // 32 is chosen because
-            // * SET_CONTAINS performs worst when smaller than but close to power of 2
-            // * Benchmark shows performance of SET_CONTAINS is better at 50, but similar at 25.
+        if (values.size() >= 10 || (type instanceof VarcharType && values.size() >= 5)) {
+            // Based on benchmark on BIGINT and DOUBLE SET_CONTAINS is already faster at 10
+            // For VARCHAR values tipping point is even sooner (at 5 elements)
```

Comment:
> SET_CONTAINS is already faster at 10

faster than ... ?

i guess than HASH_SWITCH, right?
what about DIRECT_SWITCH? if it is what it looks it is, it should be "typically the best option available"?

Reply:
> // Based on benchmark on BIGINT and DOUBLE SET_CONTAINS is already faster at 10

would it be possible to have a more "automatic benchmark", where a decision is taken _within_ the benchmark (ie benchmark using production code's decision logic, or calling a higher level API) and the result benchmarked?